### PR TITLE
fixed signup bug

### PR DESCRIPTION
**Pull request by abubakarazeem**

### CODE REVIEW

**Comment by abubakarazeem** [comment]
review: cycle 0

File: app/models/company.rb [hunk 1]
```diff
@@ -30,14 +30,14 @@ def self.current_id
   end
 
   def create_issue_states
-    IssueState.new(name: 'Resolved', company_id: id).save!
-    IssueState.new(name: 'Unresolved', company_id: id).save!
+    issue_states.create!(name: 'Resolved')
```
Comment (Anmol79) [inline app/models/company.rb]: have a constant of issue states and loop in here for creating the states

File: app/models/company.rb [hunk 2]
```diff
@@ -30,14 +30,14 @@ def self.current_id
...
+    issue_states.create!(name: 'Unresolved')
     true
   end
 
   def create_issue_types
-    IssueType.new(name: 'Improvement', company_id: id).save!
-    IssueType.new(name: 'New Feature', company_id: id).save!
+    issue_types.create!(name: 'Improvement')
```
Comment (Anmol79) [inline app/models/company.rb]: same as above

File: app/models/company.rb [hunk 3]
```diff
@@ -2,7 +2,7 @@ class Company < ActiveRecord::Base
   not_multitenant
 
   before_save { subdomain.downcase! }
-  after_save :create_issue_states, :create_issue_types
+  after_commit :create_issue_states, :create_issue_types
```
Comment (Anmol79) [inline app/models/company.rb]: after_create

File: app/models/company.rb [hunk 4]
```diff
@@ -1,8 +1,11 @@
 class Company < ActiveRecord::Base
   not_multitenant
 
+  DEFAULT_ISSUE_TYPES = %w[Improvement New\ Feature].freeze
```
Comment (Anmol79) [inline app/models/company.rb]: have the constants in respective files. like issue types should be in issue type's model

**Pull request merged**